firebase_admin/functions.py

```diff
@@ -0,0 +1,409 @@
+
+
+class TaskQueue:
+    """TaskQueue class that implements Firebase Cloud Tasks Queues functionality."""
+    def __init__(
+            self,
+            function_name: str,
+            extension_id: Optional[str],
+            project_id, credential,
+            http_client
+        ) -> None:
+
+        # Validate function_name
+        _Validators.check_non_empty_string("function_name", function_name)
+
+        self._project_id = project_id
+        self._credential = credential
+        self._http_client = http_client
+        self._function_name = function_name
+        self._extension_id = extension_id
+        # Parse resources from function_name
+        self._resource = self._parse_resource_name(self._function_name, 'functions')
+
+        # Apply defaults and validate resource_id
+        self._resource.project_id = self._resource.project_id or self._project_id
+        self._resource.location_id = self._resource.location_id or _DEFAULT_LOCATION
+        _Validators.check_non_empty_string("resource.resource_id", self._resource.resource_id)
+        # Validate extension_id if provided and edit resources depending
+        if self._extension_id is not None:
+            _Validators.check_non_empty_string("extension_id", self._extension_id)
+            self._resource.resource_id = f'ext-{self._extension_id}-{self._resource.resource_id}'
+
+
+    def enqueue(self, task_data: Any, opts: Optional[TaskOptions] = None) -> str:
+        """Creates a task and adds it to the queue. Tasks cannot be updated after creation.
+
+        This action requires `cloudtasks.tasks.create` IAM permission on the service account.
+
+        Args:
+            task_data: The data payload of the task.
+            opts: Options when enqueuing a new task (optional).
+
+        Raises:
+            FirebaseError: If an error occurs while requesting the task to be queued by
+                the Cloud Functions service.
+            ValueError: If the input arguments are invalid.
+
+        Returns:
+            str: The ID of the task relative to this queue.
+        """
+        task = self._validate_task_options(task_data, self._resource, opts)
+        service_url = self._get_url(self._resource, _CLOUD_TASKS_API_URL_FORMAT)
+        task_payload = self._update_task_payload(task, self._resource, self._extension_id)
+        try:
+            resp = self._http_client.body(
+                'post',
+                url=service_url,
+                headers=_FUNCTIONS_HEADERS,
+                json={'task': task_payload.__dict__}
+            )
+            task_name = resp.get('name', None)
+            task_resource = \
+                self._parse_resource_name(task_name, f'queues/{self._resource.resource_id}/tasks')
+            return task_resource.resource_id
+        except requests.exceptions.RequestException as error:
+            # TODO: Error handle
+            raise FirebaseError(400, 'Error', http_response=error.response)
```

Comment:
We might be able to use https://github.com/firebase/firebase-admin-python/blob/23765843a124065dcb8fd5a86a3d8f3b784df34c/firebase_admin/_utils.py#L101 here to handle the GCP errors with the correct error codes